lib/qif2json.js

```diff
@@ -44,63 +42,110 @@ function parseDate(dateStr, formats) {
   return `<invalid date:"${dateStr}">`;
 }
 
+function appendEntity(data, type, entity, currentBankName, isMultiAccount) {
+  if (isMultiAccount && currentBankName && type.list_name === 'transactions') {
+    entity.account = currentBankName;
+  }
+  if (!isMultiAccount && type.list_name === 'accounts') {
+    data.type = entity.type.replace('Type:', '');
+    return data;
+  }
+
+  if (type.list_name === 'accounts'
+      && Object.hasOwnProperty.call(data, 'accounts')
+      && data.accounts.find((a) => a.name === entity.name)
+  ) {
+    return data; // skip duplicates
+  }
+
+  if (Object.hasOwnProperty.call(data, type.list_name)) {
+    data[type.list_name].push(entity);
+  } else {
+    data[type.list_name] = [entity];
+  }
+
+  return data;
+}
+
+function clean(line) {
+  line = line.trim();
+  if (line.charCodeAt(0) === 239 && line.charCodeAt(1) === 187 && line.charCodeAt(2) === 191) {
+    line = line.substring(3);
+  }
+  return line;
+}
+
 exports.parse = function parse(qif, options) {
   /* eslint no-multi-assign: "off", no-param-reassign: "off", no-cond-assign: "off",
       no-continue: "off", prefer-destructuring: "off", no-case-declarations: "off" */
   const lines = qif.split('\n');
-  let line = lines.shift();
-  const type = /!Type:([^$]*)$/.exec(line.trim());
-  const data = {};
-  const transactions = data.transactions = [];
-  let transaction = {};
+  let type = { }; // /^(!Type:([^$]*)|!Account)$/.exec(line.trim());
+  let currentBankName = '';
+  let isMultiAccount = false;
 
-  options = options || {};
+  let data = {};
 
-  if (!type || !type.length) {
-    throw new Error(`File does not appear to be a valid qif file: ${line}`);
-  }
-  data.type = type[1];
+  let entity = {};
+
+  options = options || {};
 
   let division = {};
+  let line;
+
+  // let i = 0;
```

Comment:
Remove this comment please

Reply:
Comment removed, but i is used in error message now.